internal/resource/factory.go

```diff
@@ -64,11 +66,18 @@ func getManager() Manager {
 	if hasNVML {
 		klog.Info("Using NVML manager")
 		return NewNVMLManager()
-	} else if isTegra {
+	}
+
+	if isTegra {
```

Comment:
I would recommend first adding the mode flag so that we can properly handle the `nvml` and `tegra` modes and then rebase this PR onto those changes.